tests/postgresOpsTests.js

```diff
@@ -0,0 +1,254 @@
+ */
+
+"use strict";
+
+require("json5/lib/register");
+const fluid = require("infusion");
+const jqUnit = require("node-jqunit");
+const postgresOps = require("../src/dbOps/postgresOps.js");
+
+require("../src/shared/driverUtils.js");
+require("./shared/utilsCommon.js");
+require("./shared/utilsDb.js");
+
+// Tables structures and test data records
+require("./data/testTableModels.js");
+require("./data/testTableData.js");
+
+jqUnit.module("PostgresDB operations unit tests.");
+
+fluid.registerNamespace("fluid.tests.dbOps");
+
+const config = require("./testConfig.json5");
+
+const parameterizedInsert = `
+    INSERT INTO rgb (id, color, "colourMap") VALUES($1, $2, $3);
+`;
+
+const valueParameters = [
+    "plum",
+    "purple",
+    {
+        "name": "purple",
+        "HSL": [306, 41, 40]
+    }
+];
+
+const sqlFile = __dirname + "/data/createInsertPrefs.sql";
+
+const testData = {
+    userToInsert: `
+        INSERT INTO users
+                   ("userId",                iterations, username, name,    email,             roles,      derived_key,                                salt,                               verification_code, verified)
+            VALUES ('another.user:nonadmin', 0,          'carla',  'carla', 'carla@localhost', '{"user"}', '9ff4bc1c1846181d303971b08b65122a45174d04', '2653c80aabd3889c3dfd6e198d3dca93', 'carlaIsVerified', true)
+            RETURNING *;
+        `,
+    userToInsertJSON: {
+        "userId": "another.user:nonadmin",
+        "iterations": 0,
+        "username": "carla",
+        "name": "carla",
+        "email": "carla@localhost",
+        "roles": [
+            "user"
+        ],
+        "derived_key": "9ff4bc1c1846181d303971b08b65122a45174d04",
+        "salt": "2653c80aabd3889c3dfd6e198d3dca93",
+        "verification_code": "carlaIsVerified",
+        "verified": true
+    },
+    userChanges: `
+        UPDATE users
+            SET verified=false, email='carla@globalhost'
+            WHERE "userId"='another.user:nonadmin'
+            RETURNING *;
+    `,
+    primaryKeyChange: `
+        UPDATE users
+            SET "userId"='some.new.id'
+            WHERE "userId"='another.user:nonadmin'
+            RETURNING *;
+    `
+};
+
+const expected = {
+    userChanges: fluid.extend(
+        true, {}, testData.userToInsertJSON,
+        { verified: false, email: "carla@globalhost" }
+    ),
+    primaryKeyChange: fluid.extend(
+        true, {}, testData.userToInsertJSON,
+        { userId: "some.new.id", verified: false, email: "carla@globalhost" }
+    )
+};
+
+const rgbChartreuse = fluid.find(fluid.tests.dbOps.testTableData.rgb, function (aColour) {
+    if (aColour.id === "chartreuse") {
+        return aColour;
+    }
+});
+
+jqUnit.test("Database operations tests", async function () {
+    jqUnit.expect(234);
+
+    // Start the database
+    const dbStatus = await fluid.personalData.dockerStartDatabase(config.db.dbContainerName, config.db.dbDockerImage, config.db);
```

Comment:
These tests similarly could skip this step, perhaps controlled by an environment variable, etc.